### PR TITLE
[WFLY-18170] Use a default value for the jakarta.faces.FACELETS_BUFFE…

### PR DESCRIPTION
**Pull request by jamezp**
…R_SIZE parameter was not set.

https://issues.redhat.com/browse/WFLY-18170

### CODE REVIEW

File: jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java [hunk 1]
```diff
@@ -61,6 +86,11 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
                 JSFLogger.ROOT_LOGGER.lazyBeanValidationEnabled();
             }
         }
+        // Set a default buffer size as 1024 is too small
+        final JBossWebMetaData webMetaData = metaData.getMergedJBossWebMetaData();
+        if (!hasContextParam(webMetaData, "faces.FACELETS_BUFFER_SIZE")) {
```
Comment (bstansberry) [inline jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java:91]: The param is "jakarta.faces.FACELETS_BUFFER_SIZE". Per the spec we should also check "facelets.BUFFER_SIZE" for compatibility.

Comment (jamezp) [reply jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java:91]: Looks like a type on my side, but it also says:

> The runtime must also consider the facelets.BUFFER_SIZE param name as an alias to this param name for backwards compatibility with existing facelets tag libraries.

**Comment by bstansberry** [comment]
Thanks @jamezp

File: jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java [hunk 2]
```diff
@@ -23,6 +24,30 @@ public class JSFMetadataProcessor implements DeploymentUnitProcessor {
...
     private static final String LAZY_BEAN_VALIDATION_PARAM = "com.sun.faces.enableLazyBeanValidation";
 
+
+    private static final int defaultBufferSize;
+
+    // This is copied from org.wildfly.extension.undertow.ByteBufferPoolDefinition to come up with a decent default for
+    // the jakarta.faces.FACELETS_BUFFER_SIZE property. We calculate this because the default is 1024, which is very
+    // small, https://jakarta.ee/specifications/faces/4.0/jakarta-faces-4.0.html#a6088. Per the spec we could use -1
+    // as the default, but Mojarra does not currently support that. Once https://github.com/eclipse-ee4j/mojarra/issues/5262
+    // is resolved, and we can upgrade, we should be able to default to -1.
```
Comment (fjuma) [inline jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java:34]: @jamezp Should we create a WFLY issue to track defaulting to -1 once the fix for https://github.com/eclipse-ee4j/mojarra/issues/5262 is available in a release?

Comment (jamezp) [reply jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java:34]: @fjuma That's not a bad idea, https://issues.redhat.com/browse/WFLY-18174.

Comment (fjuma) [reply jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java:34]: Thanks @jamezp!

**Pull request merged**